### ISSUE TITLE
[Bug] Memory leak in token_to_kv_pool_allocator

### ISSUE THREAD

**Issue by JamesBrianD** [sgl-project/sglang-jax#70]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

When page_size > 1 is enabled, token_pool_allocator memory leaks or insufficient free space may occur after many requests.

### Reproduction

**launch server**

JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen-7B-Chat --trust-remote-code  --dist-init-addr=[0.0.0.0:10011](http://0.0.0.0:10011/) --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.4 --chunked-prefill-size=8192 --download-dir=/tmp --dtype=bfloat16 --jax-precompile-decode-bs-paddings 64 --max-running-requests 64 --skip-server-warmup --attention-backend=fa --jax-precompile-prefill-token-paddings 8192 --page-size=64

**do profile**

curl -X POST 'http://127.0.0.1:30000/start_profile' -d '{"output_dir": {profile_dir}, "num_steps": 5}' -H 'Content-Type: application/json'

**benchmark**

python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 10 --random-input 512 --random-output 128 --random-range-ratio 1 --warmup-requests 0

### Environment

TPU v6e * 4
